programs/token/src/lib.rs

```diff
@@ -0,0 +1,17 @@
+#![no_std]
+
+pub mod instructions;
+pub mod state;
+
+pinocchio_pubkey::declare_id!("TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA");
+
+use core::mem::MaybeUninit;
+
+const UNINIT_BYTE: MaybeUninit<u8> = MaybeUninit::<u8>::uninit();
+
+#[inline(always)]
+fn write_bytes(destination: &mut [MaybeUninit<u8>], source: &[u8]) {
+    for (d, s) in destination.iter_mut().zip(source.iter()) {
```

Comment:
did you check that this actually desugars to memcpy and not to a bunch of assignments?

Reply:
I think we probably can replace this by a `memcpy`. It is using an iteration to call `write` since each position is a `MaybeUninit<u8>` but since we know the size in advance, we might be able to use `ptr::copy_nonoverlapping` instead.